tests/test_gradient_check.py

```diff
@@ -0,0 +1,120 @@
+import unittest
+
+import numpy
+
+from chainer import cuda
+from chainer import gradient_check
+from chainer.testing import attr
+from chainer.testing import condition
+
+
+if cuda.available:
+    cuda.init()
+
+
+class NumeraicalGradientTest(unittest.TestCase):
+
+    def _f(self, x):
+        return x**2
+
+    def _df(self, x):
+        return 2*x
+
+    def setUp(self):
+        self.x = numpy.random.uniform(-1, 1, (2, 1)).astype(numpy.float32)
+        self.gy = numpy.random.uniform(-1, 1, (2, 1)).astype(numpy.float32)
+        self.f = lambda x: (self._f(x),)
+        self.df = lambda x: (self._df(x),)
+
+    def check_numerical_grad(self, f, df, x, dy):
+        dx_expect, = tuple(d*dy for d in df(x))
+        func = lambda: f(x)
+        dx_actual, = gradient_check.numerical_grad(func, (x,), (dy,))
+        gradient_check.assert_allclose(dx_expect, dx_actual)
```

Comment:
How about checking the value of `dx_actual` with simpler function like `f(x, y) = x + y`?